test/test_matchers.py

```diff
@@ -36,6 +36,87 @@ def test_pretty_print_any():
     assert str(b) == "Any(<type 'str'>)" or str(b) == "Any(<class 'str'>)"
 
 
+def test_any_such_that_equality():
```

Comment:
Please split this up into individual tests, the problem with long tests such as this with lots of asserts is that if the first assert fails the rest are not run so you don't really know how bad the situation is -maybe you broke 1 test maybe you broke 50.

The fact that your asserts are grouped into symmetric pairs suggests that maybe a parameterized test might suit them